### PR TITLE
[ fix ] : using Go backend Data , to fix build failed issue .

### PR DESCRIPTION
**Pull request by bhaskarhc**
**What this PR does / why we need it**:
-  backend data now getting form DB , 
- if build fails the pipeline show `n/a` .

screenshot from 2019-03-08 17-12-52

### CODE REVIEW

File: src/app/services/ci-dashboard.service.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { HttpClient } from "@angular/common/http";
+import { map } from "rxjs/operators";
+import { Meta,Title } from '@angular/platform-browser';
+
+@Injectable()
+export class DashboardData {
+
+    private apiurl: string;
+    constructor(private http: HttpClient, private meta:Meta,private titleService: Title) {
+        this.apiurl = "http://147.75.102.37:30002";
```
Comment (chandankumar4) [inline src/app/services/ci-dashboard.service.ts]: This api url will not work for openebs.ci, this is only for test purpose, we have to deploy the backend in the workload cluster

**Comment by chandankumar4** [comment]
If some pipeline not get triggred  it will show like image

And you have to handle the edge case also like if pipeline is not triggered it will not show any data in detail pannel

**Comment by chandankumar4** [comment]
Is the drop-down button is there for gitlab stage, if it is there, please upload that image too

File: src/app/services/ci-dashboard.service.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { HttpClient } from "@angular/common/http";
+import { map } from "rxjs/operators";
+import { Meta,Title } from '@angular/platform-browser';
+
+@Injectable()
+export class DashboardData {
+
+    private apiurl: string;
+    constructor(private http: HttpClient, private meta:Meta,private titleService: Title) {
+        this.apiurl = "http://147.75.102.37:30003";
```
Comment (chandankumar4) [inline src/app/services/ci-dashboard.service.ts]: apiurl should be: openebs.ci/api/

File: src/app/services/ci-dashboard.service.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    getEksDetails() {
```
Comment (chandankumar4) [inline src/app/services/ci-dashboard.service.ts]: Now the api changed to:- 
1- packet/v11
2 - packet/v12
3 - packet/v13

File: package.json [hunk 4]
```diff
@@ -29,13 +28,13 @@
     "dateformat": "^3.0.3",
     "font-awesome": "^4.7.0",
     "jquery": "^3.3.1",
-    "moment": "^2.22.2",
```
Comment (inyee786) [inline package.json]: where we are using this package??

Comment (bhaskarhc) [reply package.json]: @inyee786 
Here `moment` using in calcuate master build count 
**Moment package :**
> Parse, validate, manipulate, and display dates and times in JavaScript

File: src/app/services/ci-dashboard.service.ts (hunk 3 above)
Comment (bhaskarhc) [reply src/app/services/ci-dashboard.service.ts]: done :+1:

File: src/app/services/ci-dashboard.service.ts (hunk 2 above)
Comment (bhaskarhc) [reply src/app/services/ci-dashboard.service.ts]: done :+1:
